<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConfigWindow</class>
 <widget class="QMainWindow" name="ConfigWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1010</width>
    <height>772</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TAC Configuration Editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;Name:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="_name">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>The name of the configuration file</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;Author:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="_author">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>The author for the configuration file. Recommended format: biswroy &lt;biswroy@qti.qualcomm.com&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_8">
      <item>
       <widget class="QLabel" name="_platformLabel">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;Platform ID:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="_platformId">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>The platform id of the configuration as determined from the PINE CLI. Maximum allowed value for platform id is 99999.</string>
        </property>
        <property name="inputMask">
         <string notr="true"/>
        </property>
        <property name="maxLength">
         <number>6</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Preferred</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;PINE Version:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="_pineVersion">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>The version of the configuration as found from the PINE CLI output</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_8">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Preferred</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_7">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;Hardware Configuration:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="_platformConfigurationType">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>The platform type of the configuration</string>
        </property>
        <property name="text">
         <string>&lt;select a platform&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_7">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;Description: &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="_description">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>The description about the configuration file</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="_usbDescriptorLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;USB Descriptor String:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="_usbDescriptor">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label_6">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;Creation Date:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="_creationDate">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_9">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;Modification Date:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="_modificationDate">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QCheckBox" name="_resetCountEnable">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Whether to enable the reset count panel</string>
        </property>
        <property name="text">
         <string>Enable reset count</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QSplitter" name="_splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QFrame" name="_editorFrame">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>847</width>
         <height>300</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
      <widget class="ButtonEditor" name="_buttonEditor" native="true">
       <property name="enabled">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1010</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="_fileMenu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="_actionNew"/>
    <addaction name="_actionOpen"/>
    <addaction name="_actionPreview"/>
    <addaction name="_actionOpen_Script_Editor"/>
    <addaction name="_actionPINEExport"/>
    <addaction name="separator"/>
    <addaction name="_actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="separator"/>
    <addaction name="_actionContents"/>
    <addaction name="separator"/>
    <addaction name="_actionRefresh"/>
    <addaction name="separator"/>
    <addaction name="_actionAbout"/>
    <addaction name="_actionWhatsThis"/>
    <addaction name="_actionBugWriter"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>Options</string>
    </property>
    <property name="toolTipsVisible">
     <bool>false</bool>
    </property>
    <widget class="QMenu" name="_menuRestoreDefaults">
     <property name="title">
      <string>Restore defaults</string>
     </property>
     <property name="toolTipsVisible">
      <bool>false</bool>
     </property>
     <addaction name="_actionDefaultQuickSettings"/>
     <addaction name="_actionDefaultScriptAndVariables"/>
    </widget>
    <addaction name="_menuRestoreDefaults"/>
   </widget>
   <addaction name="_fileMenu"/>
   <addaction name="menuOptions"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="_actionNew"/>
   <addaction name="_actionOpen"/>
   <addaction name="_actionSave"/>
   <addaction name="separator"/>
   <addaction name="_actionPreview"/>
   <addaction name="_actionManage_Tabs"/>
   <addaction name="_actionOpen_Script_Editor"/>
   <addaction name="_actionPINEExport"/>
  </widget>
  <action name="_actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="_actionAbout">
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="_actionBugWriter">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Submit Bug Report...</string>
   </property>
  </action>
  <action name="_actionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources/TACConfigEditor.qrc">
     <normaloff>:/Save.png</normaloff>:/Save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="_actionContents">
   <property name="text">
    <string>Contents</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="_actionPreferences">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Preferences...</string>
   </property>
  </action>
  <action name="_actionOpen">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources/TACConfigEditor.qrc">
     <normaloff>:/Open.png</normaloff>:/Open.png</iconset>
   </property>
   <property name="text">
    <string>Open Configuration...</string>
   </property>
   <property name="toolTip">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="_actionPreview">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources/TACConfigEditor.qrc">
     <normaloff>:/Preview.drawio.png</normaloff>:/Preview.drawio.png</iconset>
   </property>
   <property name="text">
    <string>Preview Configuration...</string>
   </property>
   <property name="iconText">
    <string>Preview Configuration...</string>
   </property>
   <property name="toolTip">
    <string>Preview Configuration...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="_actionManage_Tabs">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources/TACConfigEditor.qrc">
     <normaloff>:/ManageTabs.png</normaloff>:/ManageTabs.png</iconset>
   </property>
   <property name="text">
    <string>Manage Tabs...</string>
   </property>
  </action>
  <action name="_actionNew">
   <property name="icon">
    <iconset resource="resources/TACConfigEditor.qrc">
     <normaloff>:/new-config.png</normaloff>:/new-config.png</iconset>
   </property>
   <property name="text">
    <string>New Configuration...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="_actionOpen_Script_Editor">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources/TACConfigEditor.qrc">
     <normaloff>:/script.png</normaloff>:/script.png</iconset>
   </property>
   <property name="text">
    <string>Open Script Editor...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="_actionRefresh">
   <property name="icon">
    <iconset resource="resources/TACConfigEditor.qrc">
     <normaloff>:/download-spinner.png</normaloff>:/download-spinner.png</iconset>
   </property>
   <property name="text">
    <string>Refresh Configurations</string>
   </property>
   <property name="toolTip">
    <string>Downloads the latest TAC Configurations</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="_actionWhatsThis">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>What's this?</string>
   </property>
   <property name="toolTip">
    <string>Enables you to understand what an items in the application does.</string>
   </property>
  </action>
  <action name="_actionPINEExport">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources/TACConfigEditor.qrc">
     <normaloff>:/PINEUpload.drawio.png</normaloff>:/PINEUpload.drawio.png</iconset>
   </property>
   <property name="text">
    <string>Export to PInE...</string>
   </property>
   <property name="toolTip">
    <string>Publish the configuration to PInE</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="_actionDefaultQuickSettings">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Quick settings</string>
   </property>
   <property name="toolTip">
    <string>Option unavailable as the quick settings table contains row(s)</string>
   </property>
  </action>
  <action name="_actionDefaultScriptAndVariables">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>QTAC script and variables</string>
   </property>
   <property name="toolTip">
    <string>Option unavailable as the script variables table contains row(s)</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ButtonEditor</class>
   <extends>QWidget</extends>
   <header>ButtonEditor.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources/TACConfigEditor.qrc"/>
 </resources>
 <connections/>
</ui>
